From 6eadd8b57051a8c1aee3cb89928eec5f37126ada Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Fri, 12 Jul 2024 21:31:40 +0300
Subject: [PATCH 10001/10003] integrate with balenaOS

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 include/configs/compulab-imx8m-plus.h | 21 +++++++++++----------
 1 file changed, 11 insertions(+), 10 deletions(-)

diff --git a/include/configs/compulab-imx8m-plus.h b/include/configs/compulab-imx8m-plus.h
index 954c5edbdd..576c9cf8b8 100644
--- a/include/configs/compulab-imx8m-plus.h
+++ b/include/configs/compulab-imx8m-plus.h
@@ -51,12 +51,13 @@
 	"scriptaddr=0x43500000\0" \
 	"kernel_addr_r=" __stringify(CONFIG_SYS_LOAD_ADDR) "\0" \
 	"bsp_script=boot.scr\0" \
-	"image=Image\0" \
+	"image=Image.gz\0" \
 	"splashimage=0x50000000\0" \
 	"console=ttymxc1,115200\0" \
-	"fdt_addr_r=0x43000000\0"			\
-	"fdto_addr_r=0x43800000\0"			\
-	"fdt_addr=0x43000000\0"			\
+	"fdt_addr_r=0x45000000\0"			\
+	"fdto_addr_r=0x45800000\0"			\
+	"fdt_addr=0x45000000\0"			\
+	"zip_addr=0x48440000\0" \
 	"boot_fdt=try\0" \
 	"fdt_high=0xffffffffffffffff\0"		\
 	"boot_fit=no\0" \
@@ -66,7 +67,7 @@
 	"mmcpart=" __stringify(CONFIG_SYS_MMC_IMG_LOAD_PART) "\0" \
 	"mmcroot=" CONFIG_MMCROOT " rootwait rw\0" \
 	"mmcautodetect=yes\0" \
-	"mmcargs=setenv bootargs console=${console} root=${mmcroot}\0 " \
+	"mmcargs=setenv bootargs console=${console} ${resin_kernel_root} ${os_cmdline}\0 " \
 	"loadbootscript=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${bsp_script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
 		"source\0" \
@@ -109,8 +110,8 @@
 		"usb_dev=0\0" \
 		"boot_part=1\0" \
 		"root_opt=rootwait rw\0" \
-		"emmc_ul=setenv iface mmc; setenv dev ${emmc_dev}; setenv part ${boot_part};" \
-		"setenv bootargs console=${console} root=${emmc_root} ${root_opt};\0" \
+		"emmc_ul=setenv iface mmc; setenv dev ${resin_dev_index}; setenv part ${resin_root_part};" \
+		"setenv bootargs console=${console} ${resin_kernel_root} ${os_cmdline};\0" \
 		"sd_ul=setenv iface mmc; setenv dev ${sd_dev}; setenv part ${boot_part};" \
 			"setenv bootargs console=${console} root=${sd_root} ${root_opt};\0" \
 		"usb_ul=usb start; setenv iface usb; setenv dev ${usb_dev}; setenv part ${boot_part};" \
@@ -118,7 +119,7 @@
 		"ulbootscript=load ${iface} ${dev}:${part} ${scriptaddr} ${bsp_script};\0" \
 		"ulrunbootscript=echo Running bootscript from ${iface} ${dev}:${part} ...; " \
 				"source ${scriptaddr}\0" \
-		"ulimage=echo loading ${image}; load ${iface} ${dev}:${part} ${loadaddr} ${image}\0" \
+		"ulimage=load ${iface} ${resin_dev_index}:${resin_root_part} ${zip_addr} boot/${image}; unzip ${zip_addr} ${loadaddr}; \0" \
 		"ulfdto=setenv fdto1file; for fdto1file in ${fdtofile}; do "\
 			    "echo loading ${fdto1file}; "\
 			    "load ${iface} ${dev}:${part} ${fdto_addr_r} ${fdto1file} "\
@@ -127,14 +128,14 @@
 			    "&& fdt apply ${fdto_addr_r};"\
 		"done; true;\0"\
 		"ulfdt=if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
-			"echo loading ${fdtfile}; load ${iface} ${dev}:${part} ${fdt_addr_r} ${fdtfile}; " \
+			"load ${iface} ${resin_dev_index}:${resin_root_part} ${fdt_addr_r} boot/${fdtfile};" \
 			"if env exists fdtofile;then "\
 					"run ulfdto; "\
 				"else "\
 					"true; "\
 				"fi; "\
 		"fi;\0" \
-		"bootlist=usb_ul sd_ul emmc_ul\0" \
+		"bootlist=emmc_ul\0" \
 		"bsp_bootcmd=echo Running BSP bootcmd ...; " \
 		"for src in ${bootlist}; do " \
 			"echo Running ${src} ...; " \
-- 
2.34.1

