From d3f9627e1563be438669195e39b60440df336762 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Fri, 12 Jul 2024 21:35:46 +0300
Subject: [PATCH 10002/10003] Run CRC32 checks on kernel image and fdt

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 include/configs/compulab-imx8m-plus.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/include/configs/compulab-imx8m-plus.h b/include/configs/compulab-imx8m-plus.h
index 576c9cf8b8..1faaf4237e 100644
--- a/include/configs/compulab-imx8m-plus.h
+++ b/include/configs/compulab-imx8m-plus.h
@@ -119,7 +119,7 @@
 		"ulbootscript=load ${iface} ${dev}:${part} ${scriptaddr} ${bsp_script};\0" \
 		"ulrunbootscript=echo Running bootscript from ${iface} ${dev}:${part} ...; " \
 				"source ${scriptaddr}\0" \
-		"ulimage=load ${iface} ${resin_dev_index}:${resin_root_part} ${zip_addr} boot/${image}; unzip ${zip_addr} ${loadaddr}; \0" \
+		"ulimage=if load ${iface} ${resin_dev_index}:${resin_root_part} ${zip_addr} boot/${image}; then unzip ${zip_addr} ${loadaddr}; run balena_kernel_load_crc_save; else false; fi; \0" \
 		"ulfdto=setenv fdto1file; for fdto1file in ${fdtofile}; do "\
 			    "echo loading ${fdto1file}; "\
 			    "load ${iface} ${dev}:${part} ${fdto_addr_r} ${fdto1file} "\
@@ -128,7 +128,7 @@
 			    "&& fdt apply ${fdto_addr_r};"\
 		"done; true;\0"\
 		"ulfdt=if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
-			"load ${iface} ${resin_dev_index}:${resin_root_part} ${fdt_addr_r} boot/${fdtfile};" \
+			"load ${iface} ${resin_dev_index}:${resin_root_part} ${fdt_addr_r} boot/${fdtfile} && run balena_fdt_load_crc_save;" \
 			"if env exists fdtofile;then "\
 					"run ulfdto; "\
 				"else "\
@@ -146,6 +146,8 @@
 			"fi; " \
 			"if run ulimage; then " \
 				"if run ulfdt; then " \
+					"run balena_kernel_load_crc_check; " \
+					"run balena_fdt_load_crc_check; " \
 					"booti ${loadaddr} - ${fdt_addr_r}; " \
 				"else " \
 					"if test ${boot_fdt} != yes; then " \
-- 
2.34.1

