From be1a8958454f25508ec0f0e616be6df171e3c2ea Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Fri, 12 Jul 2024 21:38:22 +0300
Subject: [PATCH 10003/10003] integrate with balenaOS #1

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 include/configs/compulab-imx8m-plus.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/configs/compulab-imx8m-plus.h b/include/configs/compulab-imx8m-plus.h
index 1faaf4237e..071a9b53c5 100644
--- a/include/configs/compulab-imx8m-plus.h
+++ b/include/configs/compulab-imx8m-plus.h
@@ -50,6 +50,7 @@
 	"autoload=off\0" \
 	"scriptaddr=0x43500000\0" \
 	"kernel_addr_r=" __stringify(CONFIG_SYS_LOAD_ADDR) "\0" \
+	"resin_kernel_load_addr=" __stringify(CONFIG_SYS_LOAD_ADDR) "\0" \
 	"bsp_script=boot.scr\0" \
 	"image=Image.gz\0" \
 	"splashimage=0x50000000\0" \
@@ -140,6 +141,7 @@
 		"for src in ${bootlist}; do " \
 			"echo Running ${src} ...; " \
 			"run ${src}; " \
+			"run resin_set_kernel_root; run set_os_cmdline;" \
 			"env exist boot_opt && env exists bootargs && setenv bootargs ${bootargs} ${boot_opt}; " \
 			"if run ulbootscript; then " \
 				"run ulrunbootscript; " \
-- 
2.34.1

