From e5fbe913b0e21334a2fc41a86643afde1c034e86 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Sat, 13 Jul 2024 12:44:29 +0300
Subject: [PATCH 20001/20002] balena: Fix env/mmc.c build issue

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 env/mmc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/env/mmc.c b/env/mmc.c
index 7a5836ad66..86fe37db3b 100644
--- a/env/mmc.c
+++ b/env/mmc.c
@@ -320,6 +320,7 @@ static inline int erase_env(struct mmc *mmc, unsigned long size,
 	return (n == blk_cnt) ? 0 : 1;
 }
 
+#endif /* CONFIG_CMD_SAVEENV && !CONFIG_SPL_BUILD */
 static int env_mmc_erase(void)
 {
 	int dev = mmc_get_env_dev();
@@ -363,7 +364,6 @@ fini:
 	fini_mmc_for_env(mmc);
 	return ret;
 }
-#endif /* CONFIG_CMD_SAVEENV && !CONFIG_SPL_BUILD */
 
 static inline int read_env(struct mmc *mmc, unsigned long size,
 			   unsigned long offset, const void *buffer)
-- 
2.17.1

